purchase_request/wizard/purchase_request_line_make_purchase_order.py

```diff
@@ -200,8 +200,16 @@ def _get_order_line_search_domain(self, order, item):
             ("name", "=", name),
             ("product_id", "=", item.product_id.id),
             ("product_uom", "=", vals["product_uom"]),
-            ("analytic_distribution", "=?", item.line_id.analytic_distribution),
         ]
+
+        if item.line_id.analytic_distribution:
+            analytic_account_ids = list(item.line_id.analytic_distribution.keys())
+            order_line_data.append(
+                ("analytic_distribution", "in", analytic_account_ids)
```

Comment:
I think matching the exact distribution, not just overlapping accounts is required.

Reply:
When you use `=` instead of `in`, you will see this warning.

`WARNING odoo.osv.expression: The domain term '('analytic_distribution', '=', ['65'])' should use the 'in' or 'not in' operator.`